WordPress/Sniffs/Theme/NoDeregisterCoreScriptSniff.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Sniff to check theme for deregister and register of core scripts (jquery)
+ *
+ * @category Theme
+ * @package  PHP_CodeSniffer
+ * @author   Simon Prosser <pross@pross.org.uk>
+ */
+class WordPress_Sniffs_Theme_NoDeregisterCoreScriptSniff extends WordPress_Sniffs_Functions_FunctionRestrictionsSniff {
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+				T_STRING,
+				T_CONSTANT_ENCAPSED_STRING,
+				T_DOUBLE_QUOTED_STRING,
+			   );
+
+	}//end register()
+
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+	 * @param int                  $stackPtr  The position of the current token
+	 *                                        in the stack passed in $tokens.
+	 *
+	 * @return void
+	 */
+	public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
+
+		$tokens  = $phpcsFile->getTokens();
+		$token   = $tokens[ $stackPtr ];
+
+		$content = trim( $token['content'], '"\'' );
+
+		$functions = array(
+			'wp_deregister_script',
+			'wp_register_script',
+		);
+
+		$scripts = array(
+			'jquery',
+		);
+
+		if ( in_array( $content, $functions, true ) ) {
+
+			$script = $phpcsFile->findNext( array( T_CONSTANT_ENCAPSED_STRING, T_DOUBLE_QUOTED_STRING ), $stackPtr );
```

Comment:
You may want to limit the search a bit just in case people commit weird code.
You can pass a `$local` parameter which will limit the search to the next semi-colon. or you can pass an `$end` parameter which you could determine based on the opening and closing parenthesis info.
Ref: http://pear.php.net/package/PHP_CodeSniffer/docs/latest/PHP_CodeSniffer/PHP_CodeSniffer_File.html#methodfindNext